api/src/main/java/openconsensus/trace/NoopTrace.java

```diff
@@ -0,0 +1,256 @@
+   * @return the no-op implementation of the {@code Tracer}.
+   */
+  private static Tracer getNoopTracer() {
+    return NoopTracer.INSTANCE;
+  }
+
+  // No-Op implementation of the Tracer.
+  private static final class NoopTracer extends Tracer {
+    private static final NoopTracer INSTANCE = new NoopTracer();
+
+    @Override
+    public Span getCurrentSpan() {
+      return BlankSpan.INSTANCE;
+    }
+
+    @Override
+    public Scope withSpan(Span span) {
+      return NoopScope.getInstance();
+    }
+
+    @Override
+    public Runnable withSpan(Span span, Runnable runnable) {
+      return runnable;
+    }
+
+    @Override
+    public <C> Callable<C> withSpan(Span span, Callable<C> callable) {
+      return callable;
+    }
+
+    @Override
+    public SpanBuilder spanBuilder(String spanName) {
+      return spanBuilderWithExplicitParent(spanName, getCurrentSpan());
+    }
+
+    @Override
+    public SpanBuilder spanBuilderWithExplicitParent(String spanName, @Nullable Span parent) {
+      return NoopSpanBuilder.createWithParent(spanName, parent);
+    }
+
+    @Override
+    public SpanBuilder spanBuilderWithRemoteParent(
+        String spanName, @Nullable SpanContext remoteParentSpanContext) {
+      return NoopSpanBuilder.createWithRemoteParent(spanName, remoteParentSpanContext);
+    }
+
+    @Override
+    public void recordSpanData(SpanData span) {
+      // Do nothing
+    }
+
+    private NoopTracer() {}
+  }
+
+  // Noop implementation of SpanBuilder.
+  private static final class NoopSpanBuilder extends SpanBuilder {
+    static NoopSpanBuilder createWithParent(String spanName, @Nullable Span parent) {
+      return new NoopSpanBuilder(spanName);
+    }
+
+    static NoopSpanBuilder createWithRemoteParent(
+        String spanName, @Nullable SpanContext remoteParentSpanContext) {
+      return new NoopSpanBuilder(spanName);
+    }
+
+    @Override
+    public Span startSpan() {
+      return BlankSpan.INSTANCE;
+    }
+
+    @Override
+    public void startSpanAndRun(Runnable runnable) {
+      runnable.run();
+    }
+
+    @Override
+    public <V> V startSpanAndCall(Callable<V> callable) throws Exception {
+      return callable.call();
+    }
+
+    @Override
+    public SpanBuilder setSampler(@Nullable Sampler sampler) {
+      return this;
+    }
+
+    @Override
+    public SpanBuilder setParentLinks(List<Span> parentLinks) {
+      return this;
+    }
+
+    @Override
+    public SpanBuilder setRecordEvents(boolean recordEvents) {
+      return this;
+    }
+
+    @Override
+    public SpanBuilder setSpanKind(Span.Kind spanKind) {
+      return this;
+    }
+
+    private NoopSpanBuilder(String name) {
+      Utils.checkNotNull(name, "name");
+    }
+  }
+
+  /**
+   * Returns an instance that contains no-op implementations for all the instances.
+   *
+   * @return an instance that contains no-op implementations for all the instances.
+   */
+  private static PropagationComponent getNoopPropagationComponent() {
```

Comment:
We should be consistent to not have all of these singletons. Use newNoopPropagationComponent and cache the result in the NoopTraceComponent

Reply:
In `NoopTags` we use singleton:
https://github.com/bogdandrutu/openconsensus/blob/958165da6b5175d3c00f2c41aee7e03fa645bfdb/api/src/main/java/openconsensus/tags/NoopTags.java#L73.

I think it makes sense to keep consistent here.

Reply:
https://github.com/bogdandrutu/openconsensus/pull/88/commits/d64a8d290d33ba6675daa5617ff8b06c2f34a475 removed all singleton. Will do another PR to remove it from other noop impl.